app/experimenter/docs/adrs/0008-integrating-htmx.md

```diff
@@ -0,0 +1,73 @@
+
+### Positive Consequences
+
+- All web pages using HTMX!
+- Better performance
+- Learn new tool HTMX
+
+### Negative Consequences
+
+- If new pages didn't succeed, we need to re-write those pages in React and instead of GraphQL we will be working on the transition to Rest API.
+
+## Pros and Cons of the Options
+
+### Get rid of GraphQL and transition to REST Api's
+
+We are currently dependant on the two packages mentioned above to use GraphQL in the project. For the time being (around one year)these packages were not releasing new versions which was one of the major factors of the performance, and recently they released new versions. We are highly dependent on the two packages and not getting the new version for the while worries us. All the API calls are using GraphQL and which increases the response time and decreases the performance of the platform. So the solution is to get rid of GraphQL and use REST API. This requires a lot of work for example changing Nimbus UI GraphQL call to REST API call, modifying frontend test cases and as well integration test cases, supporting backend for Rest API, and changing backend test as well
+Pros:
+
+- We can keep using React with Rest Api
+- Better performance
+- Keep using Jest for the frontend testing frameworks
+
+Cons:
+
+- Rewriting a lot of frontend test cases and frontend queries call
+- Writing and maintaining APIs which will be not used anywhere other than frontend
+
+### Designing new pages with HTMX, and keep the existing UI (React and GraphQL) and eventually move to HTMX
+
+In this approach we will be aiming to develop new pages with HTMX and keeping the existing pages in React (with GraphQL). It doesn't make sense to first change GraphQL to Rest API and then get rid of REST APIs too as we don't need any kind of API to use HTMX. Django has good support for HTMX and it avoids network overhead too. So to save time we can avoid the transition of GraphQL to Rest API and keep the GraphQL and start developing new pages in HTMX. We can keep the HTMX and React at the moment, and can slowly start the transition of pages from React to HTMX.
+Pros:
+
+- Don't need JEST framework for frontend testing
+- Easy HTMX testing
+- Better performance
+- Hybrid HTMX and React architecture
```

Comment:
it is advantageous as opposed to REST apis where everything needed to be in REST(both frontend and backend), we can have hybrid architecture, we don't need to work 100% transition now, we can work at our own pace and keep on delivering new features. This will not block us :)

Reply:
Ah so the advantage is less about the hybrid architecture, and more that this option allows us to make incremental changes instead of having to do a full REST API at once to replace GraphQL. Thanks!